# Titanic Dataset

Data yang digunakan pada use case ini dapat diunduh pada [Link ini](https://drive.google.com/open?id=15IyT1ODuDKgZb8WN6iG64hFJRZWJTSAz).

Dataset tersebut berisikan 887 data penumpang. Dengan masing-masing baris merepresentasikan data personal masing-masing penumpang.

### Load Library

In [1]:
# Import librarys
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

### No. 1 --  What is the dimension (col, row) of the data frame?

In [20]:
# Load dataset titatic.csv
df = pd.read_csv('titanic.csv')

In [21]:
# Menampilkan 5 baris pertama dari dataframe titanic
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [22]:
# Dimensi dari dataframe titanic
df.shape

(887, 8)

> Terdapat 887 data penumpang dengan detail informasi yang dijelaskan pada 8 features (column)

### No. 2 -- How to know data type of each variable?

In [23]:
# Menampilkan tipe data dari masing-masing features pada dataframe
df.dtypes

Survived                     int64
Pclass                       int64
Name                        object
Sex                         object
Age                        float64
Siblings/Spouses Aboard      int64
Parents/Children Aboard      int64
Fare                       float64
dtype: object

> Terdapat 4 features dengan tipe integer (Survived, Pclass, Siblings/Spouses Aboard, Parents/Children Aboard), 2 features dengan tipe string/object (Name, Sex), dan 2 features dengan tipe float (Age, Fare)

### No. 3 -- How many passengers survived (Survived = 1) and not-survived (Survived = 0)?

In [24]:
# Menampilkan banyaknya penumpang yang survive (survived = 1) dan tidak survive (survived = 0)
df['Survived'].groupby(df['Survived']).count()

Survived
0    545
1    342
Name: Survived, dtype: int64

> Terdapat 545 penumpang yang tidak survived (meninggal) dan 342 penumpang yang survived (mampu bertahan hidup)

### No. 4 -- How to drop column ‘Name’ from the data frame?

In [25]:
# Menghapus kolom 'Name' pada dataset titanic
df.drop(columns=['Name'], inplace=True)

In [26]:
# Melihat dataframe titanic tanpa kolom 'Name'
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  
0                        0   7.2500  
1                        0  71.2833  
2                        0   7.9250  
3                        0  53.1000  
4                        0   8.0500

### No. 5 -- Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)

In [27]:
# Menambahkan kolom family dengan formula yang telah ditentukan
# family = sibsp + parch
df['family'] = df['Siblings/Spouses Aboard'] + df['Parents/Children Aboard']
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  family  
0                        0   7.2500       1  
1                        0  71.2833       1  
2                        0   7.9250       0  
3                        0  53.1000       1  
4                        0   8.0500       0

### No. 6 -- As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for non-missing value). 


In [28]:
# Melihat apakah ada missing values atau tidak pada dataset titanic
df.isnull().sum()

Survived                   0
Pclass                     0
Sex                        0
Age                        0
Siblings/Spouses Aboard    0
Parents/Children Aboard    0
Fare                       0
family                     0
dtype: int64

In [29]:
# Menambahkan kolom age_miss sesuai dengan ketentukan yang diebrikan
# Jika nilai age = null maka age_miss = Yes
# Jika nilai age = null maka age_miss = No
age_miss = []
for i in range(len(df)):
    if np.isnan(df['Age'])[i]:
        age_miss.append('Yes')
    else:
        age_miss.append('No')
df['age_miss'] = age_miss
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  family age_miss  
0                        0   7.2500       1       No  
1                        0  71.2833       1       No  
2                        0   7.9250       0       No  
3                        0  53.1000       1       No  
4                        0   8.0500       0       No

### No. 7 -- Please fill Age missing value with means of existing Age values

In [30]:
# Menggantikan nilai age yang kosong dengan rataan dari seluruh nilai age pada dataset
df['Age'] = df['Age'].fillna(df['Age'].mean())
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  family age_miss  
0                        0   7.2500       1       No  
1                        0  71.2833       1       No  
2                        0   7.9250       0       No  
3                        0  53.1000       1       No  
4                        0   8.0500       0       No

### No. 8 -- What is the maximum passenger Age who survived from the tragedy? 

In [31]:
# Menentukan usia maksimum dari penumpang yang survive
df[df['Survived']==1]['Age'].max()

80.0

> Usia maksimum dari penumpang yang survive (mampu bertahan hidup) adalah 80 tahun

### No. 9 -- How many passengers survived from each ‘PClass’? 

In [32]:
# Menentukan berapa banyak penumpang yang survive pada masing masing kelas
df[df['Survived']==1].groupby('Pclass')['Survived'].count()

Pclass
1    136
2     87
3    119
Name: Survived, dtype: int64

> Banyaknya penumpang yang survive pada kelas 1 sebanyak 136 orang, kelas 2 sebanyak 87 orang, dan kelas 3 119 orang

### No. 10 -- How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for tttanic1 and 0.3 for titanic2 ? 


In [33]:
# Membagi data titanic dengan proporsi sebagai berikut:
# Titanic1 sebesar 70% dari keseluruhan data
titanic1 = df.sample(frac=0.7).reset_index()
titanic1.head()

index  Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0    318         0       3    male  22.0                        0   
1    269         1       3    male  25.0                        0   
2    642         1       1    male  48.0                        1   
3    756         0       3    male  23.0                        0   
4    215         1       3  female  27.0                        0   

   Parents/Children Aboard     Fare  family age_miss  
0                        0   7.2500       0       No  
1                        0   0.0000       0       No  
2                        0  76.7292       1       No  
3                        0  14.5000       0       No  
4                        0   7.9250       0       No

In [34]:
# Dimensi dari dataset titanic 1
titanic1.shape

(621, 10)

> Titanic1 terdiri atas 621 penumpang dari total 887 penumpang

In [35]:
# titanic 2 sebesar 30% dari keseluruhan dataset titanic
titanic2 = df.drop(titanic1.index)
titanic2.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
621         0       3    male  21.0                        0   
622         0       1    male  61.0                        0   
623         0       2    male  57.0                        0   
624         1       1  female  21.0                        0   
625         0       3    male  26.0                        0   

     Parents/Children Aboard     Fare  family age_miss  
621                        0  16.1000       0       No  
622                        0  32.3208       0       No  
623                        0  12.3500       0       No  
624                        0  77.9583       0       No  
625                        0   7.8958       0       No

In [37]:
titanic2.shape

(266, 9)

> Titanic2 terdiri atas 226 penumpang dari total 887 penumpang